pkg/envoy/rds/response.go

```diff
@@ -35,10 +32,7 @@ func NewResponse(cataloger catalog.MeshCataloger, proxy *envoy.Proxy, discoveryR
 	// clients on allowed routes.
 	inboundMeshTrafficPolicy := cataloger.GetInboundMeshTrafficPolicy(proxy.Identity, proxyServices)
 	if inboundMeshTrafficPolicy != nil {
-		inboundMeshRouteConfig := route.BuildInboundMeshRouteConfiguration(inboundMeshTrafficPolicy.HTTPRouteConfigsPerPort, proxy, cfg, trustDomain)
-		for _, config := range inboundMeshRouteConfig {
-			rdsResources = append(rdsResources, config)
-		}
+		builder.AddInboundMeshRouteConfig(inboundMeshTrafficPolicy.HTTPRouteConfigsPerPort, proxy, cfg, trustDomain)
```

Comment:
so for instance here, we may have separate calls:

```go
builder.SetHTTPRouteCOnfigsPerPort(configsPerPort)
.SetProxy(proxy)
.SetTrustDomain(trustDomain)
```

Note that we should not set the cfg, or any other client objects. Instead we should call the client objects here, and set the data on the builder